### PR TITLE
Feature/vite migration

### PR DESCRIPTION
**Pull request by shellyear**
Resolves: https://github.com/kbss-cvut/record-manager-ui/issues/88

Zistili jsme:
- kdyz v dist/config.js nastavis promenne tak jako by byli substituovane pomoci docker-compose.yml tak pak odtestovat docker znamena odtestovat "npm run preview"

### CODE REVIEW

**Comment by blcham** [comment]
@blcham @LaChope

File: src/components/history/HistoryPagination.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import {injectIntl} from "react-intl";
+import withI18n from "../../i18n/withI18n";
+import {ACTIONS_PER_PAGE, PAGINATION_DIRECTION } from "../../constants/DefaultConstants";
+import PropTypes from "prop-types";
+
+let HistoryPagination = (props) => (
+    <nav className="d-flex justify-content-center">
```
Comment (LaChope) [inline src/components/history/HistoryPagination.jsx:8]: I think flexbox is much better than bootstrap but for consistency with the rest of the app I assume bootstrap should be used

File: package.json [hunk 2]
```diff
@@ -70,14 +72,17 @@
     "querystring-es3": "^0.2.1",
     "redux-mock-store": "^1.5.4",
     "style-loader": "^1.1.3",
+    "vite": "^5.1.2",
+    "vite-plugin-simple-html": "^0.1.2",
```
Comment (blcham) [inline package.json:76]: Why this ?

File: package.json [hunk 3]
```diff
@@ -70,14 +72,17 @@
...
     "webpack": "^5.89.0",
     "webpack-cli": "^5.1.4",
     "webpack-config-utils": "^2.3.1",
     "webpack-dev-server": "^4.15.1"
   },
   "scripts": {
-    "start": "webpack serve --config ./webpack.config.js",
-    "build": "webpack --config ./webpack.config.js",
+    "preview": "vite preview",
+    "start": "vite",
```
Comment (blcham) [inline package.json:84]:
```suggestion
    "dev": "vite",
```

File: vite.config.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    },
+    server: {
+      port: 3000,
+    },
+    preview: {
+      port: 3000
+    },
+    plugins: [
+      pluginReact(),
+      simpleHtmlPlugin({
```
Comment (blcham) [inline vite.config.js:23]: We do not need this ... it is configured by config.js

Comment (blcham) [inline vite.config.js:23]: Can we use config.js/.ENV for configuring it?

File: vite.config.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    define: {
+      "process.env.PARSE_LINK_HEADER_MAXLEN": JSON.stringify(
+        env.PARSE_LINK_HEADER_MAXLEN
+      ),
+      "process.env.PARSE_LINK_HEADER_THROW_ON_MAXLEN_EXCEEDED": JSON.stringify(
+        env.PARSE_LINK_HEADER_THROW_ON_MAXLEN_EXCEEDED
+      ),
+      "import.meta.env.RECORD_MANAGER_API_URL": JSON.stringify(env.VITE_RECORD_MANAGER_API_URL),
+      "import.meta.env.RECORD_MANAGER_APP_TITLE": JSON.stringify(env.VITE_RECORD_MANAGER_APP_TITLE),
+      "import.meta.env.RECORD_MANAGER_DEV_SERVER_PORT": JSON.stringify(env.VITE_RECORD_MANAGER_DEV_SERVER_PORT),
```
Comment (blcham) [inline vite.config.js:43]: we want to remove that and use https://vitejs.dev/config/shared-options.html#envprefix RECORD_MANAGER_

File: src/components/history/HistoryPagination.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
```
Comment (blcham) [inline src/components/history/HistoryPagination.jsx:1]: We need history of this file and other files that were just moved.

**Comment by blcham** [comment]
Closing in favor of https://github.com/akaene/record-manager-ui/pull/4

**Pull request closed without merging**